src/contexts/ThemeContext/ThemeContextProvide.tsx

```diff
@@ -0,0 +1,31 @@
+'use client';
+import { PropsWithChildren, useEffect, useState } from 'react';
+import { ThemeContext } from './ThemeContext';
+
+export function ThemeProvider({ children }: PropsWithChildren) {
+  const [theme, setTheme] = useState<'light' | 'dark' | 'auto'>('auto');
+
+  const setNewTheme = (theme: 'light' | 'dark' | 'auto') => {
```

Comment:
Please use a `AppTheme[theme]` type here

Reply:
And please try to avoid using the same name for the parameter and the state

Reply:
Also these conditions are useless, you can simple pass setTheme to the context. TS will provide suggestions what you can pass to it and prevent you from passing something different, if TS doesn’t do it then you did something wrong with types